### PR TITLE
Ovl_En_Drs OK and Documented

### PR DESCRIPTION
**Pull request by emilybrooks**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

**Review by jpburnett — APPROVED** [review]
Very nice work with this!

File: assets/xml/objects/object_drs.xml [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
+        <Texture Name="gWeddingDressMannequinVineTex" OutName="vine" Format="rgba16" Width="32" Height="16" Offset="0x4848" />
+        <Texture Name="gWeddingDressMannequinFlowerTex" OutName="flower" Format="rgba16" Width="16" Height="16" Offset="0x4C48" />
+        <Texture Name="gWeddingDressMannequinFabricHighlightsTex" OutName="fabric_highlight" Format="i8" Width="32" Height="64" Offset="0x4E48" />
+        <Texture Name="gWeddingDressMannequinFabricSheerTex" OutName="fabric_sheer" Format="i8" Width="32" Height="32" Offset="0x5648" />
+        <Limb Name="gWeddingDressMannequinLimb" Type="Standard" EnumName="WEDDING_DRESS_MANNEQUIN_LIMB" Offset="0x5A48" />
+        <Limb Name="gWeddingDressMannequinMaskLimb" Type="Standard" EnumName="WEDDING_DRESS_MANNEQUIN_MASK_LIMB" Offset="0x5A54" />
+        <Limb Name="gWeddingDressMannequinPatternLimb" Type="Standard" EnumName="WEDDING_DRESS_MANNEQUIN_PATTERN_LIMB" Offset="0x5A60" />
+        <Skeleton Name="gWeddingDressMannequinSkel" Type="Flex" LimbType="Standard" LimbNone="WEDDING_DRESS_MANNEQUIN_NONE_LIMB" LimbMax="WEDDING_DRESS_MANNEQUIN_MAX_LIMB" EnumName="WeddingDressMannequinLimbs" Offset="0x5A78" />
```
Comment (EllipticEllipsis) [inline assets/xml/objects/object_drs.xml:19]: The limb enum is of the form `{ACTOR_DESCRIPTION}_LIMB_{LIMB_NAME}` (i.e. `Limb` is only a suffix for the data in the asset file, not the enum.

File: src/overlays/actors/ovl_En_Drs/z_en_drs.c [hunk 2]
```diff
@@ -30,36 +28,98 @@ const ActorInit En_Drs_InitVars = {
...
+static AnimationInfoS sAnimations = { &gWeddingDressMannequinIdleAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 };
+
+void EnDrs_CollisionUpdate(EnDrs* this, GlobalContext* globalCtx) {
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+void EnDrs_Setup(EnDrs* this, GlobalContext* globalCtx) {
+    s32 pad;
+    SkelAnime* temp = &this->skelAnime;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Drs/z_en_drs.c:62]: Is this temp required? It seems an unlikely thing to use a temp for.

File: src/overlays/actors/ovl_En_Drs/z_en_drs.c [hunk 3]
```diff
@@ -30,36 +28,98 @@ const ActorInit En_Drs_InitVars = {
...
+    }
+}
+
+void EnDrs_Idle(EnDrs* this, GlobalContext* globalCtx) {
+}
+
+void EnDrs_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDrs* this = THIS;
+    this->moonMaskObjBankIndex = func_8013D924(OBJECT_MSMO, globalCtx);
+    this->actionFunc = EnDrs_Setup;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Drs/z_en_drs.c:83]:
```suggestion
    EnDrs* this = THIS;

    this->moonMaskObjBankIndex = func_8013D924(OBJECT_MSMO, globalCtx);
    this->actionFunc = EnDrs_Setup;
```

File: src/overlays/actors/ovl_En_Drs/z_en_drs.c [hunk 4]
```diff
@@ -30,36 +28,98 @@ const ActorInit En_Drs_InitVars = {
...
+    EnDrs* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnDrs_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDrs* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    if (this->actor.draw) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Drs/z_en_drs.c:96]:
```suggestion
    if (this->actor.draw != NULL) {
```
Is preferable provided it matches.

File: src/overlays/actors/ovl_En_Drs/z_en_drs.c [hunk 5]
```diff
@@ -30,36 +28,98 @@ const ActorInit En_Drs_InitVars = {
...
+        SkelAnime_Update(&this->skelAnime);
+        EnDrs_CollisionUpdate(this, globalCtx);
+    }
+}
+
+void EnDrs_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnDrs* this = THIS;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Drs/z_en_drs.c:104]: Is this required? It doesn't seem to be used at all.

File: undefined_syms.txt [hunk 6]
```diff
@@ -1327,9 +1327,6 @@ D_060048B8 = 0x060048B8;
 
 // ovl_En_Drs
```
Comment (EllipticEllipsis) [inline undefined_syms.txt:1329]:
```suggestion
```
Can remove the header too now these are gone.

**Review by EllipticEllipsis — COMMENTED** [review]
Nice job, a few mostly stylistic things.

File: src/overlays/actors/ovl_En_Drs/z_en_drs.c (hunk 2 above)
Comment (emilybrooks) [reply src/overlays/actors/ovl_En_Drs/z_en_drs.c:62]: skelAnime gets put into the wrong stack pointer without it, I don't know if it's possible to do it in a cleaner way.
image

File: src/overlays/actors/ovl_En_Drs/z_en_drs.c (hunk 5 above)
Comment (emilybrooks) [reply src/overlays/actors/ovl_En_Drs/z_en_drs.c:104]: It can match with either gfxCtx or using globalCtx2, is one preferable over the other?

Comment (emilybrooks) [reply src/overlays/actors/ovl_En_Drs/z_en_drs.c:104]: empty pads don't have any effect

Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Drs/z_en_drs.c:105]: We came to the conclusion that this is a globalCtx2 in Discord.

**Review by tom-overton — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**